### PR TITLE
Actor sheet layout

### PR DESCRIPTION
**Pull request by laura-wenning**

### CODE REVIEW

File: src/components/sheets/ActorSheet.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { toJS } from "mobx";
+import { observer } from "mobx-react-lite";
+import React from "react";
+import { SheetElement } from "./SheetElement";
+
+/**
+ * Renders an actor sheet
+ */
+export const ActorSheet = observer((props: any) => {
+  React.useEffect(() => {
```
Comment (laura-wenning) [inline src/components/sheets/ActorSheet.tsx:12]: A little more commenting on this

File: src/components/sheets/SheetElement.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  SheetTextInput,
+} from "./elements";
+import { SheetCheckbox } from "./elements/Checkbox";
+import { SheetPage } from "./elements/Page";
+import { SheetPageable } from "./elements/Pageable";
+import { SheetPrefab } from "./elements/Prefab";
+import { SheetRadio } from "./elements/Radio";
+import { SheetRadioButton } from "./elements/RadioButton";
+
+export function SheetElement(props: any) {
```
Comment (laura-wenning) [inline src/components/sheets/SheetElement.tsx:22]: Comment

File: src/components/sheets/elements/Tabs.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  );
+}
+
+interface SheetTabsProps {
+  tabs: SheetTabElement[];
+  activeTab: number;
+  setActiveTab: (activeTab: number) => void;
+}
+
+export function SheetTabs(props: SheetTabsProps) {
```
Comment (laura-wenning) [inline src/components/sheets/elements/Tabs.tsx:41]: Comment

File: src/controllers/data/ActorSheetData.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+class ActorSheetDataManager extends DataManager<Partial<ActorSheetDocument>> {
+  public collection = "actorSheets";
+  public sheet: SheetController<Partial<ActorSheetDocument>>;
+
+  constructor() {
+    super("/api/actor-sheets");
+    this.sheet = new SheetController();
+  }
+
+  public loadSheet(key: string, xml?: string) {
```
Comment (laura-wenning) [inline src/controllers/data/ActorSheetData.ts:19]: Comment

File: src/controllers/data/ActorSheetData.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    if (xml) {
+      this.sheet.load(key, xml);
+      return;
+    }
+  }
+
+  public getPage(key: string) {
```
Comment (laura-wenning) [inline src/controllers/data/ActorSheetData.ts:26]: Comment

Comment (laura-wenning) [reply src/controllers/data/ActorSheetData.ts:26]: Rename to getSheet

File: src/controllers/layout/parser.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { isServer } from "@owl-factory/utilities/client";
+
+let domParser: any = null;
+
+export function parseXML(xml: string) {
```
Comment (laura-wenning) [inline src/controllers/layout/parser.ts:5]: Comment

File: src/types/layouts/backgroundElement.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+import { GenericSheetElement } from "./genericElement";
+
+export interface BackgroundElement extends GenericSheetElement {
```
Comment (laura-wenning) [inline src/types/layouts/backgroundElement.ts:3]: Comment all of the different elements. Also rename them?

**Review by lucyawrey — APPROVED** [review]
Looks great, 7 to go

**Pull request merged**